components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -1630,21 +1625,48 @@ public static String mapDigestAlgorithm(Algorithm signatureAlgorithm) throws Ide
     /**
      * Generic Signing function
      *
-     * @param jwtClaimsSet contains JWT body
+     * @param jwtClaimsSet       contains JWT body
      * @param signatureAlgorithm JWT signing algorithm
-     * @param tenantDomain tenant domain
+     * @param tenantDomain       tenant domain
+     * @param clientSecret
      * @return signed JWT token
      * @throws IdentityOAuth2Exception
      */
-    public static JWT signJWT(JWTClaimsSet jwtClaimsSet, JWSAlgorithm signatureAlgorithm, String tenantDomain)
+    public static JWT signJWT(JWTClaimsSet jwtClaimsSet, JWSAlgorithm signatureAlgorithm,
+                              String tenantDomain, String clientSecret)
+            throws IdentityOAuth2Exception {
+
+        if (JWSAlgorithm.RS256.equals(signatureAlgorithm) || JWSAlgorithm.RS384.equals(signatureAlgorithm) ||
+                JWSAlgorithm.RS512.equals(signatureAlgorithm)) {
+            return signJWTWithRSA(jwtClaimsSet, signatureAlgorithm, tenantDomain);
+        } else if (JWSAlgorithm.HS256.equals(signatureAlgorithm) || JWSAlgorithm.HS384.equals(signatureAlgorithm) ||
+                JWSAlgorithm.HS512.equals(signatureAlgorithm)) {
+            return signWithHMAC(jwtClaimsSet, signatureAlgorithm, clientSecret);
+        } else {
+            // return signWithEC(jwtClaimsSet,jwsAlgorithm,request); implementation need to be done
+            throw new RuntimeException("Provided signature algorithm: " + signatureAlgorithm +
+                    " is not supported");
+        }
+    }
+
+    /**
+     * Generic Signing function
+     *
+     * @param jwtClaimsSet       contains JWT body
+     * @param signatureAlgorithm JWT signing algorithm
+     * @param tenantDomain       tenant domain
+     * @return signed JWT token
+     * @throws IdentityOAuth2Exception
+     */
+    public static JWT signJWT(JWTClaimsSet jwtClaimsSet, JWSAlgorithm signatureAlgorithm,
+                              String tenantDomain)
             throws IdentityOAuth2Exception {
 
         if (JWSAlgorithm.RS256.equals(signatureAlgorithm) || JWSAlgorithm.RS384.equals(signatureAlgorithm) ||
                 JWSAlgorithm.RS512.equals(signatureAlgorithm)) {
             return signJWTWithRSA(jwtClaimsSet, signatureAlgorithm, tenantDomain);
         } else if (JWSAlgorithm.HS256.equals(signatureAlgorithm) || JWSAlgorithm.HS384.equals(signatureAlgorithm) ||
                 JWSAlgorithm.HS512.equals(signatureAlgorithm)) {
-            // return signWithHMAC(jwtClaimsSet,jwsAlgorithm,request); implementation need to be done
```

Comment:
Why not referring to hmac singing from here as well

Reply:
This is to where to use in utill where HMAC is not applicable implementation mainly covers the client_id signing